src/compiler/checker.ts

```diff
@@ -3308,6 +3306,15 @@ namespace ts {
             return getSignaturesOfObjectOrUnionType(getApparentType(type), kind);
         }
 
+        function typeHasConstructSignatures(type: Type): boolean {
+            let apparentType = getApparentType(type);
+            if (apparentType.flags & (TypeFlags.ObjectType | TypeFlags.Union)) {
+                let resolved = resolveObjectOrUnionTypeMembers(<ObjectType>type);

```

Comment:
This is what the existing `typeHasCallOrCosntructSignatures` does below, this just checks for construct signatures only.